### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

One test left to fix, would really benefit from tracing to debug.
This only runs on `hardhat`

Fixes # (issue)

## Checklist

- [ ] I have run vyper and solidity linting
- [ ] I have run the tests on my machine
- [ ] I have followed commitlint guidelines
- [ ] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

**Comment by fubuloubu** [comment]
> Got one failing conftest that prevents one test to run:
> 
> ```
> AttributeError: 'ProjectManager' object has no attribute 'ZapClaim'
> ```
> 
> Not sure what's going on with ape here @fubuloubu

Hmm, that is pretty bizarre. Does it work locally and only fails in CI? If it fails locally, can you do `ape compile -f` and then it works? Think it might be something to do with build caching logic

**Comment by fubuloubu** [comment]
> It's possible that this has been solved but a new version of ape-foundry is required. [ApeWorX/ape-foundry@345bced](https://github.com/ApeWorX/ape-foundry/commit/345bced85d7f052d5cda36b1d0b3792ddeb94ba0)

Can release `ape-foundry` once https://github.com/ApeWorX/ape-foundry/pull/5 is merged

File: contracts/RewardPool.vy [hunk 1]
```diff
@@ -144,14 +142,14 @@ def _find_timestamp_epoch(ve: address, _timestamp: uint256) -> uint256:
 
 @view
 @internal
-def _find_timestamp_user_epoch(ve: address, user: address, _timestamp: uint256, max_user_epoch: uint256) -> uint256:
+def _find_timestamp_user_epoch(user: address, _timestamp: uint256, max_user_epoch: uint256) -> uint256:
```
Comment (banteg) [inline contracts/RewardPool.vy:145]: i think these utilities could be exposed from veyfi? that would also be cheaper and less code in this contract.

File: contracts/VotingYFI.vy [hunk 2]
```diff
@@ -336,6 +336,10 @@ def withdraw() -> Withdrawn:
 
     return Withdrawn({amount: old_locked.amount - penalty, penalty: penalty})
 
+@view
+@external
+def e_find_epoch_by_block(user: address, height: uint256, max_epoch: uint256) -> uint256:
```
Comment (banteg) [inline contracts/VotingYFI.vy:341]: this should be called `find_epoch_by_block`, the internal function should be called `_find_epoch_by_block`

File: contracts/interfaces/IVotingYFI.sol [hunk 3]
```diff
@@ -14,7 +14,11 @@ interface IVotingEscrow is IERC20 {
 
     function locked(address) external view returns (LockedBalance memory);
 
-    function deposit_for(address, uint256) external;
+    function modify_lock(
+        uint256,
+        uint256,
+        address
```
Comment (banteg) [inline contracts/interfaces/IVotingYFI.sol:20]: add argument names

**Review by banteg — APPROVED** [review]
good to merge, feel free to address the comments in a new pull request

**Pull request merged**